internal/femto/femto.go

```diff
@@ -0,0 +1,68 @@
+// Package femto provides a tiny abstraction layer for writing API's that serve JSON over HTTP.
+package femto
+
+import (
+	"encoding/json"
+	"log/slog"
+	"net/http"
+	"sync/atomic"
+)
+
+type Femto struct {
+	mux   http.ServeMux
+	reqNo atomic.Uint64
+}
+
+func (f *Femto) ListenAndServe(addr string) error {
+	return http.ListenAndServe(addr, &f.mux)
+}
+
+func OnPost[T any](f *Femto, pattern string, handle HandlerFunc[T]) {
+	f.mux.HandleFunc(pattern, func(w http.ResponseWriter, r *http.Request) {
+		doPost(f, w, r, handle)
+	})
+}
+
+func doPost[T any](f *Femto, w http.ResponseWriter, r *http.Request, handle HandlerFunc[T]) {
+
+	reqNo := f.nextReqNo()
+	log := f.logger().With(slog.Uint64("req_no", reqNo))
+
+	log.Info("New Request", "method", r.Method, "url", r.URL, "from", r.RemoteAddr)
+
+	if r.Method != http.MethodPost {
+		log.Info("Wanted GET, returned 405")
+		http.Error(w, "Expected GET", http.StatusMethodNotAllowed)
```

Comment:
Might be my 3am brain, but are we expecting gets or posts here?

Reply:
No, this is my 3am brain